app/src/main/java/com/greenfox/kalendaryo/adapter/KalendarAdapter.java

```diff
@@ -43,14 +68,78 @@ public void addKalendarResponse(List<GetKalendarResponse> newGetKalendarResponse
 
     @Override
     public void onBindViewHolder(KalendarAdapter.ViewHolder holder, int position) {
-        GetKalendarResponse getKalendarResponse = getKalendarResponses.get(position);
+        DaggerApiComponent.builder().build().inject(this);
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                popUpWindow(view, position);
+            }
+        });
+        kalendarSetText(holder, position);
+        }
+
+    @Override
+    public int getItemCount() {
+        return kalendarResponses.size();
+    }
+
+    public void removeAt(int position) {
+        kalendarResponses.remove(position);
+        notifyItemRemoved(position);
+    }
+
+    public void kalendarSetText(KalendarAdapter.ViewHolder holder, int position){
+        GetKalendarResponse getKalendarResponse = kalendarResponses.get(position);
         holder.kalendarDescription.setText(getKalendarResponse.getOutputCalendarId());
         holder.kalendarName.setText(getKalendarResponse.getOutputGoogleAuthId());
     }
 
-    @Override
-    public int getItemCount() {
-        return getKalendarResponses.size();
+    public void popUpWindow(View view, int position){
+        AlertDialog.Builder alert = new AlertDialog.Builder(view.getContext());
+        alert.setTitle("Delete Kalendar");
+        alert.setMessage("Are you sure to delete the kalendar?");
+        alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialogInterface, int i) {
+                kalPref = new KalPref(view.getContext());
+                String clientToken = kalPref.clientToken();
+
+                backendApi.deleteKalendar(clientToken, kalendarIdToDelete(position)).enqueue(new Callback<Void>() {
```

Comment:
do not put a method call as a method parameter, save the kalendarID into a variable before the backend api call